lib/index.js

```diff
@@ -164,3 +166,40 @@ const findNodesForFilter = () => {
 const removeFoodRow = () => {
   event.target.closest('article').remove()
 }
+
+const calorieSorter = () => {
+  var originalOrder = []
+  var count = 0
+
+  $('strong.calorie-sort').on('click', event => {
+    if (originalOrder.length === 0) {
+      var order = $('div.all-foods-list article.food-item-row')
+      order.each(index => {
+        originalOrder.push(order[index])
+      })
+    }
+    count += 1
+    var calorieNodes = $('div.all-foods-list article.food-item-row')
+    sortBy(calorieNodes, count, originalOrder)
+  })
+}
+
+const sortBy = (calorieNodes, clickCount, originalOrder) => {
+  if (clickCount === 1 || clickCount % 3 === 1) {
+    var sorted = calorieNodes.sort((a, b) => {
+      return findCalorieCount(a) - findCalorieCount(b)
+    })
+    $('div.all-foods-list').append(sorted)
+  } else if (clickCount === 2 || clickCount % 3 === 2) {
+    var sorted = calorieNodes.sort((a, b) => {
+      return findCalorieCount(b) - findCalorieCount(a)
+    })
+    $('div.all-foods-list').append(sorted)
+  } else {
+    $('div.all-foods-list').append(originalOrder)
```

Comment:
I like the use of preserving the article nodes in the order of which they were created/added to a variable instead of doing a get request to reset the table. This works much more smoothly and the user won't experience the lag of seeing the table empty and then repopulated.